### PR TITLE
[FLINK-15558][Connector] Bump Elasticsearch version from 7.3.2 to 7.5.1 for es7 connector

### PR DESCRIPTION
**Pull request by yanghua**
## What is the purpose of the change

*This pull request bumps Elasticsearch version from 7.3.2 to 7.5.1 for es7 connector*

## Brief change log

  - *Bump Elasticsearch version from 7.3.2 to 7.5.1 for es7 connector*
  - *Update NOTICE FILE*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**yes** / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by yanghua** [comment]
Relevant test error. rebuild now.

**Comment by yanghua** [comment]
@flinkbot run travis

**Comment by yanghua** [comment]
Hi @aljoscha Can you please review this PR? I have also updated the NOTICE file.

File: flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -15,31 +15,33 @@ This project bundles the following dependencies under the Apache Software Licens
...
+- org.apache.lucene:lucene-suggest:8.3.0
+- org.elasticsearch:elasticsearch:7.5.1
+- org.elasticsearch:elasticsearch-cli:7.5.1
+- org.elasticsearch:elasticsearch-core:7.5.1
+- org.elasticsearch:elasticsearch-secure-sm:7.5.1
+- org.elasticsearch:elasticsearch-x-content:7.5.1
+- org.elasticsearch.client:elasticsearch-rest-client:7.5.1
+- org.elasticsearch.client:elasticsearch-rest-high-level-client:7.5.1
+- org.elasticsearch.plugin:aggs-matrix-stats-client:7.5.1
+- org.elasticsearch.plugin:lang-mustache-client:7.5.1
```
Comment (aljoscha) [inline flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/NOTICE]: Why did you add `mustache-client` here? I think the rest of the dependencies look correct but about this one I don't know. It doesn't appear in the list of jars that are shaded in when I run `mvn clean install -DskipTests -Dfast`. In the ES7 SQL connector module.

Comment (yanghua) [reply flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/NOTICE]: I run `mvn dependency:tree` and it shows this plugin.

Comment (aljoscha) [reply flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/NOTICE]: But it's not in the resulting jar. If you run `mvn clean install -DskipTests -Dfast` in the ES7 SQL module you will get some output like this:

```
[INFO] Including org.apache.flink:flink-connector-elasticsearch7_2.11:jar:1.11-SNAPSHOT in the shaded jar.
[INFO] Including org.apache.flink:flink-connector-elasticsearch-base_2.11:jar:1.11-SNAPSHOT in the shaded jar.
[INFO] Including org.elasticsearch.client:elasticsearch-rest-high-level-client:jar:7.5.1 in the shaded jar.
[INFO] Including org.elasticsearch:elasticsearch:jar:7.5.1 in the shaded jar.
[INFO] Including org.elasticsearch:elasticsearch-core:jar:7.5.1 in the shaded jar.
[INFO] Including org.elasticsearch:elasticsearch-secure-sm:jar:7.5.1 in the shaded jar.
[INFO] Including org.elasticsearch:elasticsearch-x-content:jar:7.5.1 in the shaded jar.
[INFO] Excluding org.yaml:snakeyaml:jar:1.17 from the shaded jar.
[INFO] Including com.fasterxml.jackson.core:jackson-core:jar:2.10.1 in the shaded jar.
[INFO] Including com.fasterxml.jackson.dataformat:jackson-dataformat-smile:jar:2.8.11 in the shaded jar.
[INFO] Including com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.8.11 in the shaded jar.
[INFO] Including com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:jar:2.8.11 in the shaded jar.
[INFO] Excluding org.elasticsearch:elasticsearch-geo:jar:7.5.1 from the shaded jar.
[INFO] Including org.apache.lucene:lucene-core:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-analyzers-common:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-backward-codecs:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-grouping:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-highlighter:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-join:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-memory:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-misc:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-queries:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-queryparser:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-sandbox:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-spatial:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-spatial-extras:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-spatial3d:jar:8.3.0 in the shaded jar.
[INFO] Including org.apache.lucene:lucene-suggest:jar:8.3.0 in the shaded jar.
[INFO] Including org.elasticsearch:elasticsearch-cli:jar:7.5.1 in the shaded jar.
[INFO] Excluding net.sf.jopt-simple:jopt-simple:jar:5.0.2 from the shaded jar.
[INFO] Excluding com.carrotsearch:hppc:jar:0.8.1 from the shaded jar.
[INFO] Excluding joda-time:joda-time:jar:2.5 from the shaded jar.
[INFO] Excluding com.tdunning:t-digest:jar:3.2 from the shaded jar.
[INFO] Excluding org.hdrhistogram:HdrHistogram:jar:2.1.9 from the shaded jar.
[INFO] Excluding org.elasticsearch:jna:jar:4.5.1 from the shaded jar.
[INFO] Including org.elasticsearch.client:elasticsearch-rest-client:jar:7.5.1 in the shaded jar.
[INFO] Including org.apache.httpcomponents:httpclient:jar:4.5.3 in the shaded jar.
[INFO] Including org.apache.httpcomponents:httpcore:jar:4.4.6 in the shaded jar.
[INFO] Including org.apache.httpcomponents:httpasyncclient:jar:4.1.4 in the shaded jar.
[INFO] Including org.apache.httpcomponents:httpcore-nio:jar:4.4.12 in the shaded jar.
[INFO] Including commons-codec:commons-codec:jar:1.10 in the shaded jar.
[INFO] Including commons-logging:commons-logging:jar:1.1.3 in the shaded jar.
[INFO] Including org.elasticsearch.plugin:mapper-extras-client:jar:7.5.1 in the shaded jar.
[INFO] Including org.elasticsearch.plugin:parent-join-client:jar:7.5.1 in the shaded jar.
[INFO] Including org.elasticsearch.plugin:aggs-matrix-stats-client:jar:7.5.1 in the shaded jar.
[INFO] Including org.elasticsearch.plugin:rank-eval-client:jar:7.5.1 in the shaded jar.
[INFO] Excluding org.elasticsearch.plugin:lang-mustache-client:jar:7.5.1 from the shaded jar.
[INFO] Excluding com.github.spullara.mustache.java:compiler:jar:0.9.6 from the shaded jar.
[INFO] Including org.apache.logging.log4j:log4j-to-slf4j:jar:2.11.1 in the shaded jar.
[INFO] Including org.apache.logging.log4j:log4j-api:jar:2.11.1 in the shaded jar.
[INFO] Including org.apache.flink:force-shading:jar:1.11-SNAPSHOT in the shaded jar.
[INFO] No artifact matching filter io.netty:netty
```
which tells you what is included in the jar.

Comment (yanghua) [reply flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/NOTICE]: OK, Got it. I referenced the dependency tree before. Will remove it now.

```
- org.elasticsearch.plugin:lang-mustache-client:7.5.1
```

Comment (yanghua) [reply flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/NOTICE]: Done.

Comment (aljoscha) [reply flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/NOTICE]: Thanks, I think this is good to merge now! 🍵
